### PR TITLE
WARN: Clarify future warnings for read_csv

### PR DESCRIPTION
**Pull request by phofl**
- [x] Ensure all linting tests pass, see [here](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit) for how to run them

In other places we emit a specific warning message what to use in the future. I find this quite helpful for the user.

### CODE REVIEW

File: pandas/io/parsers/readers.py [hunk 1]
```diff
@@ -446,10 +446,10 @@
     "low_memory",
 }
 
-_deprecated_defaults: dict[str, Any] = {
-    "error_bad_lines": None,
-    "warn_bad_lines": None,
-    "squeeze": None,
+_deprecated_defaults: dict[str, tuple[Any, str]] = {
+    "error_bad_lines": (None, "Use on_bad_lines in the future."),
```
Comment (mroeschke) [inline pandas/io/parsers/readers.py:450]: Could the `None`s just be replaced with the error message and the downstream code refactored appropriately?

Comment (phofl) [reply pandas/io/parsers/readers.py:450]: The None initializes the default value of the argument. We have to kepe it, if we want to keep the structure generic for the future.

Comment (mroeschke) [reply pandas/io/parsers/readers.py:450]: Gotcha. A comment describing a format, or use of a Namedtuple, would be great to clarify that here.

Comment (phofl) [reply pandas/io/parsers/readers.py]: Added a namedtuple, makes the access easiert too

**Review by mroeschke — APPROVED** [review]

**Comment by mroeschke** [comment]
Thanks @phofl

**Pull request merged**